torch/csrc/jit/codegen/cuda/ir_utils.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/type.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+namespace ir_utils {
+
+template <typename FilterType, typename Iterator>
+class FilterIterator {
+ public:
+  FilterIterator(Iterator begin, Iterator end) : current_(begin), end_(end) {
+    advance();
+  }
+
+  FilterType* operator*() const {
+    return (*current_)->template as<FilterType>();
+  }
+
+  FilterType* operator->() const {
+    return (*this);
+  }
+
+  FilterIterator& operator++() {
+    ++current_;
+    advance();
+    return *this;
+  }
+
+  FilterIterator operator++(int) {
+    auto before_increment = *this;
+    ++current_;
+    advance();
+    return before_increment;
+  }
+
+  bool operator==(const FilterIterator& other) const {
+    TORCH_INTERNAL_ASSERT(
+        end_ == other.end_,
+        "Comparing two FilteredViews that originate from different containers");
+    return current_ == other.current_;
+  }
+
+  bool operator!=(const FilterIterator& other) const {
+    return !(*this == other);
+  }
+
+ private:
```

Comment:
please place the private state section last (public methods, private methods, private state)

Reply:
Done